contracts/Hub.sol

```diff
@@ -0,0 +1,123 @@
+        ISyntheticAsset(tickersymbolToSynthAssetContractAddress[tickerSymbol]).mint(receiver, amount); 
+    }
+
+    /**
+     * @notice sets the contract address of on sythetic asset (e.g synthTSLA)
+     * @param _synthAssetAddress contract address
+     */
+    function setSynthAssetContractAddress(address _synthAssetAddress , string memory tickerSymbol) public onlyOwner{
+        tickersymbolToSynthAssetContractAddress[tickerSymbol] = _synthAssetAddress ;
+    }
+
+    /**
+     * @notice sets the Fund contract address
+     * @param _fundAddress address of Fund contract
+     */
+    function setCollateralFundsContract(address _fundAddress) public onlyOwner{
+        fundContract = ICollateralFunds(_fundAddress);
+    }
+
+    /**
+     * @notice sets the Storage contract address
+     * @param _storageAddress address of Storage contract
+     */
+    function setStorageContract(address _storageAddress) public onlyOwner{
+        storageContract = IStorage(_storageAddress);
+    }
+
+    /**
+     * @notice checks the collateral ratio of an address.
+     * @dev for all assets combined. No individual / isolated positions for now.
+     * @dev price checking of colllateral too -> volatile collateral / depegging of stable collateral
+     * @param addr address of user whom collateral ratio is to be checked.
+     * @param collateralTickerSymbol identifier of token used for collateral#
+     * @return collateral ratio
+     */
+    function getCollateralRatioByAddress(address addr, string memory collateralTickerSymbol) public view returns (uint256) {
+        //Check amount funded
+        uint256 amountFunded = fundContract.getCollateralByAddress(msg.sender);
+        uint256 collateralPrice = storageContract.getAssetPrice(collateralTickerSymbol);
+        uint256 collateralValue = amountFunded * collateralPrice;
+
+        //Check assets minted
+        string[] memory assetsMinted = storageContract.getAssetListOfUser(addr);
+        // Total value of minted assets
+        uint256 totalValueMinted = 0;
+        //Sum up total value of minted assets
+        for(uint i = 0; i < assetsMinted.length; i++) {
+            uint256 assetAmount = storageContract.getAssetAmountOfUser(addr, assetsMinted[i]);
+            uint256 assetPrice = storageContract.getAssetPrice(assetsMinted[i]);
+            uint256 assetValue = assetAmount * assetPrice;
+            totalValueMinted += assetValue;
+        }
+        //return collateral / totalValueMinted 
+        return collateralValue  / totalValueMinted;
+    }
+
+    function createTradingPairOnUniswap(string memory tickerSymbol) external onlyOwner {
+        uint24 fee = 3000;
+        address syntheticContract = tickersymbolToSynthAssetContractAddress[tickerSymbol];
+        address newTradingPool = factory.createPool(syntheticContract, USDCKovan, fee);
+        tickerSymbolToTradingPool[tickerSymbol] = newTradingPool;
+    }
+}
+
+interface ISyntheticAsset {
```

Comment:
you changed the contract `SyntheticAsset` to `Factory` this should be updated then